plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/actions/TabbarArrangeMenuManager.java

```diff
@@ -163,36 +166,79 @@ private void refreshArrangeMenu(IWorkbenchPage page, ISelection selection) {
             }
         }
 
```

Comment:
Looks good but I am not sure to follow. You commented that you handle some case where the menu is empty or out of date. Is it a bug that you are fixing along with your move pinned elements feature or I am missing something?

Reply:
I've simply reorganized the code to avoid duplication. This avoids having to add common actions to the diagram and diagram elements twice (in `if (arrangeSelection)` and in `else`).